<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>620</width>
    <height>799</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DBC Editor for CAN Bus Database Files</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>621</width>
      <height>781</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="whatsThis">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Edit</string>
     </attribute>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>230</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Bit Start</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_name">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>250</y>
        <width>231</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QTextEdit" name="CAN_ID">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Name</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>290</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Min</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sg_length">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>250</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_offset">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>310</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>410</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Debug</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>80</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Comment</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_max">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>310</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>230</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Type</string>
      </property>
     </widget>
     <widget class="QTextBrowser" name="DBC_temp">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>520</y>
        <width>511</width>
        <height>201</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextBrowser" name="debug">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>430</y>
        <width>511</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QPushButton" name="bt_addDBC">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>650</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Add DBC</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>290</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Unit</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="msg_cm">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>511</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_factor">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>310</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="bt_signal">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>560</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Add Signal</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="msg_name">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>100</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QComboBox" name="sg_type">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>250</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Signed</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Unsigned</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>CAN ID</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>290</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Factor</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>290</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Max</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>230</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Endian</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>290</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Offset</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>500</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>DBC File</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>200</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Signal</string>
      </property>
     </widget>
     <widget class="QPushButton" name="bt_del_lastline">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>600</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Delete
Last Line</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>230</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Message</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sg_start">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>250</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QSpinBox" name="msg_length">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>100</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>80</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Name</string>
      </property>
     </widget>
     <widget class="QPushButton" name="bt_msg">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>520</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Add Message</string>
      </property>
     </widget>
     <widget class="QPushButton" name="bt_save">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>690</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Save DBC</string>
      </property>
     </widget>
     <widget class="QComboBox" name="sg_endian">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>250</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Little</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Big</string>
       </property>
      </item>
     </widget>
     <widget class="QTextEdit" name="sg_cm">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>370</y>
        <width>511</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_unit">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>310</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>350</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Comment</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_min">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>310</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_22">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>10</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>19</pointsize>
       </font>
      </property>
      <property name="text">
       <string>DBC Edit</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Read</string>
     </attribute>
     <widget class="QTextBrowser" name="txt_DBCfilepath">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>511</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>DBC File Path</string>
      </property>
     </widget>
     <widget class="QPushButton" name="bt_OpenDBC">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>30</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Open</string>
      </property>
     </widget>
     <widget class="QTextBrowser" name="txt_debug_readtab">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>511</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_23">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>65</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Debug</string>
      </property>
     </widget>
     <widget class="QPushButton" name="bt_Clear">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>90</y>
        <width>81</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Clear</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_26">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>145</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Msg List</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_27">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>360</y>
        <width>61</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Sg List</string>
      </property>
     </widget>
     <widget class="QComboBox" name="sg_select_combo">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>360</y>
        <width>69</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QTableWidget" name="sg_table">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>390</y>
        <width>601</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QTableWidget" name="msg_table">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>601</width>
        <height>181</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Live</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_mode1">
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>30</y>
        <width>71</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Mode 1</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_mode2">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>30</y>
        <width>71</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Mode 2</string>
      </property>
     </widget>
     <widget class="QComboBox" name="sg_select_combo_2">
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>60</y>
        <width>69</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextBrowser" name="txt_debug_readtab_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>441</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_24">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Data</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_25">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>5</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Debug</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_close">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>60</y>
        <width>71</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Close</string>
      </property>
     </widget>
     <widget class="QTableWidget" name="live_table">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>601</width>
        <height>241</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
